src/legacy/ui/public/saved_objects/helpers/confirm_modal_promise.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiConfirmModal } from '@elastic/eui';
+import { npStart } from 'ui/new_platform';
+import { toMountPoint } from '../../../../../plugins/kibana_react/public';
+
+export function confirmModalPromise(message = '', title = '', confirmBtnText = ''): Promise<true> {
+  return new Promise((resolve, reject) => {
+    const cancelButtonText = i18n.translate(
+      'common.ui.savedObjects.confirmModal.cancelButtonLabel',
+      {
+        defaultMessage: 'Cancel',
+      }
+    );
+
+    const modal = npStart.core.overlays.openModal(
```

Comment:
Can we pass the overlays down via parameters here as well?

Reply:
done!